### DISCUSSION TITLE
How to Pass Data to External Data Connector REST Query in Automation?

### DISCUSSION THREAD

**Discussion by omar-ehab** [Budibase/budibase#10248] [Help] [answered]
Hey there! I'm trying to set up automation triggered by the ROW CREATED event, and I want to send an API Request using an external data connector action. I've created a REST API data source and selected a query, but I'm not sure how to pass data from the ROW CREATED action to that query in the data source. Any ideas or suggestions would be greatly appreciated!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @omar-ehab,

You will need to setup a binding before hand, I've done a very simple version of this using a public api. (See below image)
<img width="983" alt="binding setup" src="https://user-images.githubusercontent.com/126772285/231141080-33d0e860-3dbf-4981-982f-524708c3f1ac.png">

Once thats done you should be able to go back to the automation and see the binding which you created within the external connector. From here you can attach whatever field value you want from the created row.
<img width="507" alt="Automation creation" src="https://user-images.githubusercontent.com/126772285/231141544-3ea3e211-5edc-4a0c-8dd0-67f19853288d.png">

Input below
<img width="426" alt="input" src="https://user-images.githubusercontent.com/126772285/231141741-3b25f5de-10ac-4a72-9b36-8f32b43cc797.png">

Output below
<img width="424" alt="output" src="https://user-images.githubusercontent.com/126772285/231141763-8791a7b5-0eb0-4528-97a6-a54cb6b1c391.png">

I hope this helps!
